core/dbt/adapters/base/impl.py

```diff
@@ -1328,7 +1328,8 @@ def render_raw_columns_constraints(cls, raw_columns: Dict[str, Dict[str, Any]])
         rendered_column_constraints = []
 
         for v in raw_columns.values():
-            rendered_column_constraint = [f"{v['name']} {v['data_type']}"]
+            col_name = cls.quote(cls, v["name"]) if v.get("quote") else v["name"]
```

Comment:
could we do something similar for model-level constraints in [`render_model_constraint`](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/adapters/base/impl.py#L1400) to avoid the user having to specify `columns` with quoting (e.g. `columns: [ '"from"' ]`)

Reply:
I think this would require us to:
- Look up the `columns` dict for this model — which doesn't seem in the scope of `render_model_constraint`, or passed in as an argument
- Match up the column names specified in the `columns` list with the keys of the `columns` dict, to determine if a column entry by the same name has `quote: true`

So - might be possible? I'm not convinced it's worth the extra effort